src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreUtils.cs

```diff
@@ -108,9 +111,26 @@ public static async Task<Stream> CopyRemoteStreamLocallyAsync(Stream stream, Rav
 
             try
             {
-                AssertFreeSpace(stream, basePath.FullPath);
+                AssertFreeSpace(size, basePath.FullPath);
+
+                var sw = Stopwatch.StartNew();
+                var swForProgress = Stopwatch.StartNew();
+                long totalRead = 0;
+
+                onProgress?.Invoke($"Copying ZipArchive locally, size: {size}");
+
+                stream.CopyTo(file, readCount =>
```

Comment:
you need to create CopyToAsync extension

Reply:
done